Sources/Sentry/SentryFramesTracker.m

```diff
@@ -1,14 +1,25 @@
 #import "SentryFramesTracker.h"
 #import "SentryDisplayLinkWrapper.h"
+#import "SentryNSArrayRingBuffer.h"
+#import "SentryProfilingConditionals.h"
+#import "SentryTracer.h"
 #import <SentryLog.h>
 #import <SentryScreenFrames.h>
 #include <stdatomic.h>
 
 #if SENTRY_HAS_UIKIT
 #    import <UIKit/UIKit.h>
 
+/**
+ * A ring-buffered version of @c SentryFrameTimestampInfo so we can build the results here and limit
+ * how many timestamps we will retain.
+ */
+typedef SentryNSArrayRingBuffer<NSDictionary<NSString *, NSNumber *> *>
+    SentryMutableFrameTimestampInfo;
+
 static CFTimeInterval const SentryFrozenFrameThreshold = 0.7;
 static CFTimeInterval const SentryPreviousFrameInitialValue = -1;
+static NSUInteger const SentryNumberOfFrameTimestampsToRetain = 10000;
```

Comment:
`l`: How did we come up with 10000? Can we add the info as a comment, please?

Reply:
I think it's still an open question as to how many we should limit this to; I just made what I thought was a conservative guess. 

A previous suggestions was 5 minutes' worth if we recorded all frame timestamps. If we only record slow/frozen frames, and all frames render just slow enough to be considered slow on a 120 Hz device, this would be 5*60*119=35,700 frames (the 119 comes from where we calculate the slow frame threshold to be the reciprocal of the ideal frame rate minus 1 frame).

Reply:
After some further discussion we think we should not limit the number of entries here. The rationale is that the profiler runs at 100Hz and generates orders of magnitude more data than the timestamps we're recording for dropped frames, which would be capped at 119 Hz for current high end devices. If we have concerns about payload size, we should look into limiting duration of profiles first.